### ISSUE TITLE
improve comment stripping

### ISSUE THREAD

**Issue by klevstul** [klevstul/kDeployer#4]
logic for stripping away js comments is too simple. corner cases / special cases fail.

**Comment by klevstul** [comment]
test of comment parsing:
`>perl kDeployer.pl -e prod -min bar -f comment.bar`

**Comment by klevstul** [comment]
the solution was found here:
http://perldoc.perl.org/perlfaq6.html#How-do-I-use-a-regular-expression-to-strip-C-style-comments-from-a-file%3f

`s#/\*[^*]*\*+([^/*][^*]*\*+)*/|//([^\\]|[^\n][\n]?)*?\n|("(\\.|[^"\\])*"|'(\\.|[^'\\])*'|.[^/"'\\]*)#defined $3 ? $3 : ""#gse;`

i tried this as well, but never worked for me:
```
    s{
              /\*         ##  Start of /* ... */ comment
              [^*]*\*+    ##  Non-* followed by 1-or-more *'s
              (
                  [^/*][^*]*\*+
              )*          ##  0-or-more things which don't start with /
                                      ##    but do end with '*'
              /           ##  End of /* ... */ comment
          |         ##     OR  various things which aren't comments:
              (
                  "           ##  Start of " ... " string
                  (
                      \\.           ##  Escaped char
                  |               ##    OR
                      [^"\\]        ##  Non "\
                  )*
                  "           ##  End of " ... " string
              |         ##     OR
                  '           ##  Start of ' ... ' string
                  (
                      \\.           ##  Escaped char
                  |               ##    OR
                      [^'\\]        ##  Non '\
                  )*
                  '           ##  End of ' ... ' string
              |         ##     OR
                  .           ##  Anything other char
                  [^/"'\\]*   ##  Chars which doesn't start a comment, string or escape
              )
          }{defined $2 ? $2 : ""}gxse;
```

**Comment by klevstul** [comment]
unfortunately it did not do this trick, after all. this code is causing problems:
```
    var /** @type {?} */ XSSI_PREFIX = /^\)\]\}',?\n/;
    /**
     * Creates connections using `XMLHttpRequest`. Given a fully-qualified
     * request, an `XHRConnection` will immediately create an `XMLHttpRequest` object and send the
     * request.
     *
     * This class would typically not be created or interacted with directly inside applications, though
     * the {\@link MockConnection} may be interacted with in tests.
     *
     * \@experimental
     */
    var XHRConnection = (function () {
        /**
         * @param {?} req
         * @param {?} browserXHR
         * @param {?=} baseResponseOptions
         */
        var response = new Response(responseOptions);
        response.ok = isSuccess(status);
        if (response.ok) {
            responseObserver.next(response);
            // TODO(gdi2290): defer complete if array buffer until done
            responseObserver.complete();
            return;
        }
        responseObserver.error(response);
        if (!req.headers.has('Accept')) {
            req.headers.append('Accept', 'application/json, text/plain, */*');
        }
```

closing this issue though, and will try to fix this with #5.
